src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -165,8 +165,18 @@ internal static async Task<NameDeclarationInfo> GetDeclarationInfoAsync(Document
                     _ => ImmutableArray.Create(
                         new SymbolKindOrTypeKind(SymbolKind.Local),
                         new SymbolKindOrTypeKind(MethodKind.LocalFunction)),
-                    cancellationToken);
-                return result.Type != null;
+                    cancellationToken,
+                    out var expression);
+
+                if (result.Type is null || expression is null)
+                    return false;
+
+                // we have something like `x.y $$`.
+                //
+                // For this to actually be the start of a local declaration or function x.y needs to bind to an actual
+                // type symbol, not just any arbitrary expression that might have a type (e.g. `Console.BackgroundColor $$).
+                var symbol = semanticModel.GetSymbolInfo(expression, cancellationToken).GetAnySymbol();
```

Comment:
I do wonder if we can simply not provide name suggestions for dotted expressions. Get it's symbol might not be cheap and this pattern is relatively rare.

Reply:
we def want name completions.  For example:

```
Foo.Customer $$
```

Would want to suggest `customer` here :)

This entire type is already heavily symbol-based.  prior to this it was computing the tyep of the expression anyways, so just ensureing that we compute its symobl (not type) is fine from a perf perspective.